Training A-B potential with different datasets(A, B, and AB)

**Summary**

<!--Please provide a clear and concise description of what the question is.-->

If I want to train a potential of A-B, could I ***use different datasets* with the varied elements(for example, *A, B, AB*)**?

**Deepmd-kit Version, Python Version, Tensorflow Version, GCC Version and Cuda Version.**

<!--Please specify precisely which Deepmd-kit version this issue was detected with on what build environment, (Python Version and GCC Version are in the first and second line of Python command line interface respectively. And check the Deepmd-kit Version with command line "print(deepmd.__version__)"). -->


- deepmd-kit (2.0.0b4.dev10+g1f01650d)
- python  (3.7.3)
- tensorflow (2.3.0)
- gcc (Debian 8.3.0-6) 
- cuda (none)     


<!--If applicable, specify what platform you are running on. -->

**Details**

<!--Please explain the issue in detail here-->

For this case, A-dataset only includes A element while AB-dataset includes both A and B elements.
After running `dp train input.json`, I got an error:
```
DEEPMD INFO    ---Summary of DataSystem: training     -----------------------------------------------
DEEPMD INFO    found 12 system(s):
DEEPMD INFO              system               natoms  bch_sz   n_bch   prob  pbc
DEEPMD INFO    -- AB/system/dataset      32       1             193      0.114    T
DEEPMD INFO    --  A/system/dataset      32       1             190      0.110    T
DEEPMD INFO    --------------------------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/lgy/.local/bin/dp", line 8, in <module>
    sys.exit(main())
  File "/home/lgy/.local/lib/python3.7/site-packages/deepmd/entrypoints/main.py", line 419, in main
    train_dp(**dict_args)
  File "/home/lgy/.local/lib/python3.7/site-packages/deepmd/entrypoints/train.py", line 216, in train
    _do_work(jdata, run_opt)
  File "/home/lgy/.local/lib/python3.7/site-packages/deepmd/entrypoints/train.py", line 266, in _do_work
    model.build(train_data, stop_batch)
  File "/home/lgy/.local/lib/python3.7/site-packages/deepmd/train/trainer.py", line 290, in build
    assert (self.ntypes >= data.get_ntypes()), "ntypes should match that found in data"
AssertionError: ntypes should match that found in data
```

In my input.json, I only modified a little things on [***hackathon2021 Cu example***](https://github.com/deepmodelinghackathon/hackathon2021/blob/master/A1/input.json)
```
model/type_map  : ["A", "B"]

training/training_data/systems: [
    "path/to/A-dataset",
    "path/to/AB-dataset" ]
training/training_data/batch_size: "auto"
```

Can deepmd-kit train a A-B potential in this way? if can, how to fix my `input.json`?

Thanks for any help.

--- a/deepmd/train/trainer.py
+++ b/deepmd/train/trainer.py
@@ -295,7 +295,16 @@ class DPTrainer (object):
             # Usually, the type number of the model should be equal to that of the data
             # However, nt_model > nt_data should be allowed, since users may only want to 
             # train using a dataset that only have some of elements 
-            assert (self.ntypes >= data.get_ntypes()), "ntypes should match that found in data"
+            if self.ntypes < data.get_ntypes():
+                raise ValueError(
+                    "The number of types of the training data is %d, but that of the "
+                    "model is only %d. The latter must be no less than the former. "
+                    "You may need to reset one or both of them. Usually, the former "
+                    "is given by `model/type_map` in the training parameter (if set) "
+                    "or the maximum number in the training data. The latter is given "
+                    "by `model/descriptor/sel` in the training parameter." % (
+                        data.get_ntypes(), self.ntypes
+                ))
             self.type_map = data.get_type_map()
             self.batch_size = data.get_batch_size()
             self.model.data_stat(data)

